### PR TITLE
Autolathe fixes and code improvements

### LINKED ISSUE #78000: Autolathes mass print stackable items
Reporting client version: 514.1589

## Reproduction:
Print a batch of anything stackable from an autolathe like cable coil or material sheets or even conveyor belts

watch in horror as it spams out 110 sheets when you only wanted 15

### PR DESCRIPTION
**Pull request by SyncIt21**
## About The Pull Request
1. Fixes #78000
The while loop ran 1 too many times when printing stacks
2. Autolathe prints stacks of any amount in constant time as was the case before
3. The autolathe will award the client with `getting_an_upgrade` only once when printing multiple items that does not contain iron or glass & updates the UI immediately upon printing

## Changelog
:cl:
fix: autolathe prints the correct amount of stacks and merges it with existing stacks on the turf
fix: autolathe takes constant time when printing stacks
code: autolathe will attempt to award you the "getting an upgrade" award only once when printing multiple items & updates UI immediately
/:cl:

### CODE REVIEW

**Comment by mc-oofert** [comment]
balls

**Comment by SyncIt21** [comment]
> balls

Shit happens

File: code/game/machinery/autolathe.dm [hunk 1]
```diff
@@ -311,30 +311,37 @@
...
 			if(length(picked_materials))
 				new_item.set_custom_materials(picked_materials, 1 / multiplier) //Ensure we get the non multiplied amount
-				for(var/x in picked_materials)
-					var/datum/material/M = x
-					if(!istype(M, /datum/material/glass) && !istype(M, /datum/material/iron))
-						user.client.give_award(/datum/award/achievement/misc/getting_an_upgrade, user)
-		if(drop_direction) //no need to call if ontop of us
+				if(!client_awarded) //so we dont award the medal multiple times
+					for(var/x in picked_materials)
+						var/datum/material/M = x
```
Comment (Time-Green) [inline code/game/machinery/autolathe.dm:333]: I know it’s not your code, but named param would be cool 🙂

**Review by Time-Green — APPROVED** [review]

**Review by ArcaneMusic — APPROVED** [review]
Looks clean

**Pull request merged**